### ISSUE TITLE
Axios Post Works but Responds with Network Error

### ISSUE THREAD

**Issue by DanJ210** [axios/axios#1827]
I searched for this here, around the internet and I went and signed up in the chat but no one's there. No response in a day.

#### Summary

I'm learning so excuse my newb code. I'm using async functions with axios that all work except for one. This one did work before as I never saw this error come up when I implemented it. The `updateItem(item)` method is the one giving an error. What's weird is that I was using `axios.put` no problem but was getting an invalid verb error. Switched to `axios.post` and it does work, it hits the method and sends the item and when the method is finished an Ok(200) response is returned. But an error is thrown on the `axios.post` call of a network error. I'll show the code and error below.

```js
async function deleteSelectedItemsAsync(removedItems) {
	try {
		let response = await axios.post('/Items/DeleteItems', removedItems);
		
		console.log('Delete response: ' + response.status);
	} catch (e) {
		console.log(e);
	}
}

async function updateItem(item) {
	var response;
	try {
		item.editButtonMessage = "Edit";
		response = await axios.post('/Items/UpdateItems', item);
		
	} catch (e) {
		console.log(response);
		console.log(e);
	}
}

async function addItemToDatabaseAsync(item) {
	try {
		const response = await axios.post('/Items/PostItem', item);
	} catch (e) {
		alert(e);
	}
}
```
![axios_error](https://user-images.githubusercontent.com/16159685/46885327-9d1f8300-ce25-11e8-95c0-a46f8f4aaa7d.PNG)

#### Context

- axios version: v0.18.0
- Environment: Windows 10, Edge  41.16299.666.0, Chrome Dev 71.0.3573.0 (Official Build) dev (64-bit), and Regular Chrome latest version

**Comment by EduardoSaverin** [comment]
Okay! Where does this axios Instance Point?

**Comment by DanJ210** [comment]
I never did get a response in the provided chat room. Any idea here?

**Comment by DanJ210** [comment]
@EduardoSaverin Anything else I can provide? Axios is included with a `<script>` reference and then I just use `axios.post` or `get` or whatever.

**Comment by DanJ210** [comment]
:(

**Comment by allarddehaas** [comment]
@DanJ210 I get the error when my API is not accepting CORS.
Do you POST from localhost to an API?

**Comment by DanJ210** [comment]
I will try the suggested steps tomorrow. To answer @allarddehaas, I'm posting from localhost when developing. I'll check and confirm that localhost is the only time I'm getting the error, not on the published site.

Thanks for the responses. Will report what happens.

**Comment by DanJ210** [comment]
I haven't been able to work on what I've wanted to recently but now I've been able to get back to this. I've changed the `try/catch` block situation to use the regular `.then().catch()` and I'm getting the same `Network Error` issue still except developer tools is showing me something different than above. I'm not POSTing to an API but I am running from localhost. I'm posting to an MVC endpoint that's in the same application so it has the same URL which I suppose can be called an API but there should be no cross-origin there.

Below is my code, I tried `error.response.data` @heyamykate but I'm told it's undefined. If I just log out `console.log(error)` I get `Error: Network Error` exactly. But browser developer tools will tell me `net::ERR_INCOMPLETE_CHUNKED_ENCODING 200 (OK)` which is weird that it has the `200 (OK)` part in it.

So the `axios.post()` does work, the endpoint it calls does get hit and the data is sent, the task is performed and then the endpoint returns an `Ok(200)` result. But I still get this error which the catch block catches.

```
axios
    .post('/Items/UpdateItems', item)
    .then(function(result) {
      console.log('Logging result ' + result);
    })
    .catch(function(error) {
      console.log('What happened? ' + error.response.data);
    });
```

**Comment by ervivek2512** [comment]
do one thing. change url to the page where is just show string and then check still it comes in catch block. if not then there is some problem here '/Items/UpdateItems'.

**Comment by DanJ210** [comment]
I'm not sure that I follow, @ervivek2512.

**Comment by brettpthomas** [comment]
@DanJ210 Not sure if this helps, but I had a similar issue that was caused by axios calling a different port on localhost. 

I had a selenium test running on http://localhost:51641, but erroneously POSTing to a development server on http://localhost:8000. :8000 returned a `200`, but axios threw a `Error: Network Error`.

**Comment by fraserdale** [comment]
Having the same error on two local host ports. When looking in dev tools I can see the request returns correct contents but showing CORS error and Network Error

**Comment by LucienShui** [comment]
Having the same error :(

**Comment by ibovangeffen** [comment]
Having the same issue. Postman does work, axios doesn't

**Comment by jasonsaayman** [comment]
This sounds a lot like a mixture of errors. I doubt the error lies within Axios though from what I have read. If someone can give an example where it is shown to be a definitive Axios issue please open a new ticket for that issue.

**Comment by incentum** [comment]
I ran into an error similar to this. Very baffling. It turned out to be cors related. The options works, and the post works, or at least gets a 200 in dev tools, but you still get a "Network Error..." thrown. It's because the "Access-Control-Allow-Origin" header is missing on the post request. Works fine if you add that.

**Comment by ManiruzzamanAkash** [comment]
In my case, after adding the cors in server, at the php code, I've made this error disappear. Here is the different language system to enable cors - 
https://enable-cors.org/server.html

**Comment by nicolas15000** [comment]
> 
> 
> I ran into an error similar to this. Very baffling. It turned out to be cors related. The options works, and the post works, or at least gets a 200 in dev tools, but you still get a "Network Error..." thrown. It's because the "Access-Control-Allow-Origin" header is missing on the post request. Works fine if you add that.

Hello, could you please describe how you added "Access-Control-Allow-Origin"  to the axios post ? Thank you

**Comment by nicolas15000** [comment]
Hello, please help me : 
'NETWORK ERROR':
https://stackoverflow.com/questions/58717673/axios-sending-network-error-while-status-is-200-ok

**Comment by ahaq0** [comment]
Any luck?

**Comment by djflorio** [comment]
Not sure how helpful this is for everyone else, but I was experiencing the exact same issue with the login route of my own project. The request succeeded and login worked, but `axios` threw a `networkError`. I also tried the request in Postman, where it worked flawlessly.

For me the issue ended up being in the API itself, _not_ `axios`. Turns out I had a misspelled function within the login route in the backend that I was calling with `await`, and for whatever reason it failed completely silently. It was pretty deeply nested within other `async/await` functions, which could explain why it was so hard to track down. Either way, fixing the misspelling also fixed the issue I was having with `axios` on the frontend.

TL;DR: your issue might be in the route you're calling, not `axios`. Check the backend code for any silly mistakes if you haven't already.

**Comment by kenberkeley** [comment]
Maybe... Disable your AD blockers in your browser? ....
https://github.com/axios/axios/issues/383#issuecomment-575408350

**Comment by ihaseebkhan** [comment]
Having same issue, any solutions yet?

**Comment by martinduo** [comment]
Having the same issue. Postman does work, axios doesn't. I wrote both server side and client side.
Server reduced to this dead simple code;

const httpTrigger: AzureFunction = async function(context: Context, req: HttpRequest): Promise<void> {
  if (req.body && req.body.name) {
    context.res = {
      body: { hello: 'bruce' }
    }
  } else {
    context.res = { status: 400, body: '{"Error":"Bad input."}',}
  }
}

Client doing this request:

  let objJson = JSON.stringify(planObject)
  let url = 'http://localhost:7071/api/MoneyPlan'
  let res = await axios.post(url, objJson)

Both client and server running in debug mode (VSCode) on my machine. Client sends Post, Server receives message and returns 200 ok. Client alerts Network Error.  This was working a few weeks ago.

**Comment by julioacosta** [comment]
I've found the solution here. If you are testing on android device at localhost, localhost is 10.0.2.2
https://github.com/axios/axios/issues/973#issuecomment-437221047

**Comment by martinduo** [comment]
Well in my case the problem persisted even when running client and server online with both the web site and server function hosted on Azure. I fixed that problem by using fetch instead of Axios. But even with fetch, I still had the problem running local tests. Right now I'm busy on other things, but I'll try your suggestion later.

**Comment by erwinwahyura** [comment]
Having same issue axios version: `0.19.0`, any solutions yet?

**Comment by Oscar-Silva** [comment]
Same issue here :[

**Comment by Strateus** [comment]
Same issue, happens only after browser idles for more than an hour and an access_token expires, so this should throw a 401:
'{"message":"The incoming token has expired"}'

my API (on AWS) is CORS-enabled, and it has 2 methods: OPTIONS and POST. 
request to OPTIONS:

> Method request headers: {Origin=http://localhost:3333, Accept=*/*, Referer=http://localhost:3333/dashboard, User-Agent=Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.130 Safari/537.36, X-Forwarded-Proto=https, Sec-Fetch-Site=cross-site, Host=<https rest endpoint removed>, Accept-Encoding=gzip, deflate, br, Pragma=no-cache, X-Forwarded-Port=443, X-Amzn-Trace-Id=Root=1-5e4d2d03-a2c3c0ccea66dbdcf2158fd3, Sec-Fetch-Mode=cors, Access-Control-Request-Method=POST, Cache-Control=no-cache, Access-Control-Request-Headers=authorization,content-type,x-api-key, X-Forwarded-For=69.124.173.195, Accept-Language=ru-RU,ru;q=0.9,en-US;q=0.8,en;q=0.7}

OPTIONS returns this:

> Method response headers: {Access-Control-Allow-Headers=Content-Type,X-Amz-Date,Authorization,X-Api-Key,X-Amz-Security-Token, Access-Control-Allow-Methods=OPTIONS,POST, Access-Control-Allow-Origin=*, Content-Type=application/json}
> Successfully completed execution
> Method completed with status: 200

Then i have this in my browser console:

> Error: Network Error
>     at createError (http://localhost:3333/static/js/0.chunk.js:5077:15)
>     at XMLHttpRequest.handleError (http://localhost:3333/static/js/0.chunk.js:4572:14)
![image](https://user-images.githubusercontent.com/5196895/74837756-33a06600-52f0-11ea-8d9e-5a8a2bc04fca.png)

Then later in the stack:

![image](https://user-images.githubusercontent.com/5196895/74837853-677b8b80-52f0-11ea-863e-8831cfc4a2bb.png)
(last line is related to my throw after i have not found any response in the error object)

Request is correct and works from with client (tried requests.posts in python)
Any hints?

**Comment by niceangular** [comment]
Same issue here!!
![image](https://user-images.githubusercontent.com/48355589/75328289-be68fe00-58b8-11ea-8c4c-3f8b845e066f.png)

**Comment by zarlo12** [comment]
Try with 

`<?php
header('Access-Control-Allow-Origin: *');
header("Access-Control-Allow-Headers: X-API-KEY, Origin, X-Requested-With, Content-Type, Accept, Access-Control-Request-Method");
header("Access-Control-Allow-Methods: GET, POST, OPTIONS, PUT, DELETE");
header("Allow: GET, POST, OPTIONS, PUT, DELETE");`

**Comment by ssayebee** [comment]
![image](https://user-images.githubusercontent.com/51493975/76095257-524c6f80-6007-11ea-8b3d-7d46c15f5905.png)

I have same problem
It works in Postman, but not when a request is sent to axios.

axios version `^0.19.2`

```javascript
  const handleUpload = event => {
    const data = new FormData();
    data.append("image", event.target.files[0])
    
    api.post("images", data)
    .then(response => console.log(response))
    .catch(errors => console.log(errors));
  }
```

**Comment by Bilaltcpaas** [comment]
I was getting **Network Error** when making an API request using AXIOS only in Window Server Firefox.Following are the steps to fix the issue.I am using NodeJS server

1-  npm install cors

2-const cors = require("cors");

3- app.use(cors());

4- app.use(function(req, res, next) {
  res.setHeader("Access-Control-Allow-Origin", "*");
  res.setHeader("Access-Control-Allow-Credentials", "true");
  res.setHeader(
    "Access-Control-Allow-Methods",
    "GET,HEAD,OPTIONS,POST,PUT,DELETE"
  );
  res.setHeader(
    "Access-Control-Allow-Headers",
    "Origin,Cache-Control,Accept,X-Access-Token ,X-Requested-With, Content-Type, Access-Control-Request-Method"
  );
  if (req.method === "OPTIONS") {
    return res.status(200).end();
  }
  next();
});

**Comment by Lachlann** [comment]
@Bilaltcpaas I was having this issue using Mockoon adding the headers defined above resolved the issue for me, thank you.

**Comment by yosukeonrails** [comment]
if youre trying to reach localhost, just replace the localhost with `10.0.2.2`

https://stackoverflow.com/questions/5528850/how-do-you-connect-localhost-in-the-android-emulator

**Comment by Strateus** [comment]
So in my case the problem was the following:
1. user auth token expired, and browser sent a regular request to backend (OPTIONS + POST)
2. backend responded with 200 correctly on OPTIONS request, with correct CORS headers
3. then axios sent a second request to main method (in my case it is POST)
4. since token have expired - backend returned 401, but **axios expects CORS headers to be present with this 401 response as well** (see [here ](https://stackoverflow.com/questions/49886315/axios-interceptors-response-undefined) and [here](https://github.com/serverless/serverless/issues/3191)), but backend have not been providing them (i did not know i need to configure CORS headers on AWS API Gateway 4XX responses in addition to regular CORS configuration for every REST Endpoint)

After i added CORS headers to 4xx processing - everything worked well.

**Comment by tobiasfeil** [comment]
It would be nice if axios said something about CORS if the error stems from CORS.

**Comment by shotaK** [comment]
It does say that it is a CORS issue, but in Firefox, in Chrome yes it does not show what is wrong.
